.ci/Jenkinsfile

```diff
@@ -28,70 +28,107 @@ pipeline {
+        */
+        stage('Checkout') {
+          steps {
+            pipelineManager([ cancelPreviousRunningBuilds: [ when: 'PR' ] ])
+            deleteDir()
+            gitCheckout(basedir: "${BASE_DIR}")
+            stash allowEmpty: true, name: 'source', useDefaultExcludes: false
           }
         }
-      }
-    }
-    /**
-     Run the test suite.
-     */
-    stage('Tests') {
-      steps {
-        cleanup()
-        withMageEnv(){
-          withKubernetes() {
-            withCloudTestEnv() {
+        /**
+        Check the source code.
+        */
+        stage('Check') {
+          steps {
+            cleanup()
+            withMageEnv(){
               dir("${BASE_DIR}"){
-                sh(label: 'Check',script: 'make test check-git-clean')
+                sh(label: 'Check',script: 'make check-static')
               }
             }
           }
         }
-      }
-      post {
-        always {
-          dir("${BASE_DIR}") {
-            archiveArtifacts(allowEmptyArchive: true, artifacts: 'build/test-results/*.xml')
-            archiveArtifacts(allowEmptyArchive: true, artifacts: 'build/kubectl-dump.txt')
-            archiveArtifacts(allowEmptyArchive: true, artifacts: 'build/elastic-stack-dump/stack/logs/*.log')
-            archiveArtifacts(allowEmptyArchive: true, artifacts: 'build/elastic-stack-dump/check/logs/*.log')
-            junit(allowEmptyResults: false,
-                keepLongStdio: true,
-                testResults: "build/test-results/*.xml")
-            coverageReport('build/test-coverage')
+        /**
+        Run the unit tests suite
+        */
+        stage('Unit tests') {
+          steps {
+            cleanup()
+            withMageEnv(){
+              dir("${BASE_DIR}"){
+                sh(label: 'Check',script: 'make test-go-ci')
+              }
+            }
+          }
+          post {
+            always {
+              dir("${BASE_DIR}") {
+                archiveArtifacts(allowEmptyArchive: true, artifacts: 'build/test-results/*.xml')
+                junit(allowEmptyResults: false,
+                    keepLongStdio: true,
+                    testResults: "build/test-results/*.xml")
+                coverageReport('build/test-coverage')
+              }
+            }
           }
         }
-      }
-    }
-    stage('Release') {
-      when {
-        tag pattern: '(v)?\\d+\\.\\d+\\.\\d+', comparator: 'REGEXP'
-      }
-      steps {
-        withMageEnv(){
-          withCredentials([string(credentialsId: "${GITHUB_TOKEN_CREDENTIALS}", variable: 'GITHUB_TOKEN')]) {
-            dir("${BASE_DIR}") {
-              sh 'curl -sL https://git.io/goreleaser | bash'
+        stage('Integration Tests') {
+          failFast true
+          options { skipDefaultCheckout() }
+          steps {
+            withGithubNotify(context: 'Integration Tests', tab: 'tests') {
+              cleanup()
+              dir("${BASE_DIR}") {
+                script {
```

Comment:
I struggled a bit to see what is this script executing at the end. Could this be moved to a function that receives the `testCommands` as parameter?

Reply:
Or seeing https://github.com/elastic/elastic-package/pull/485/files#r690968480, could this be moved to a function for each command? So something like this is done here to chose the different options for each command:
```
parallel([
  'stack-command': integrationTest(command: 'test-stack-command'),
  'check-packages': integrationTest(command: 'test-check-packages`, junitArtifacts: true, publishCoverage: true),
  'profiles-command': integrationTest(command: 'test-profiles-command'),
])
```

Reply:
Done in 58c4f66